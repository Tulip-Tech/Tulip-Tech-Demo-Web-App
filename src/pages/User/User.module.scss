@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

.user {
  display: flex;
  justify-content: center;
  margin: variables.$base-spacing * 3.5 variables.$base-spacing * 4;
  color: var(--primary-color);
  font-family: var(--body-alt-font-family);

  @include responsive.mobile-only() {
    margin: 0 variables.$base-spacing;
  }

  @include responsive.tablet-only() {
    margin: 0 variables.$base-spacing * 1.5;
  }
}
.leftColumn {
  width: 250px;
  margin-right: 24px;
  padding-left: 19px;

  font-weight: var(--body-font-weight-bold);
  font-size: 18px;
  font-style: normal;
  line-height: 20px;
  letter-spacing: 0.5px;
}

.button {
  margin-bottom: variables.$base-spacing;
}

.logoutLi {
  margin-bottom: 0;
  padding-top: variables.$base-spacing;
  border-top: 1px solid rgba(255, 255, 255, 0.32);
  > a {
    margin-bottom: 0;
  }
}

.mainColumn {
  width: 100%;
  max-width: 750px;
}

.panel {
  margin-bottom: variables.$base-spacing * 1.5;
  padding: variables.$base-spacing;
  font-weight: normal;
  font-size: 16px;
  font-style: normal;
  line-height: 18px;
  letter-spacing: 0.15px;
  background: theme.$panel-bg;
  box-shadow: theme.$panel-box-shadow;

  > ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
  }
}

.panelHeader {
  margin-bottom: variables.$base-spacing;
  padding-bottom: variables.$base-spacing;
  border-bottom: theme.$panel-header-border-bottom;

  > h3 {
    font-weight: var(--body-font-weight-bold);
    font-size: 24px;
    line-height: 26px;
    text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);
  }
}

.checkbox {
  display: flex;
  align-items: center;
  > input {
    margin-right: 10px;
  }
}
