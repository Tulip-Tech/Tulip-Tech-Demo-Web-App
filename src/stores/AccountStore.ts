import type { PaymentDetail, Subscription, Transaction } from '#types/subscription';
import type { AuthData, Consent, Customer, CustomerConsent } from '#types/account';
import { createStore } from '#src/stores/utils';

type AccountStore = {
  loading: boolean;
  auth: AuthData | null;
  user: Customer | null;
  subscription: Subscription | null;
  transactions: Transaction[] | null;
  activePayment: PaymentDetail | null;
  customerConsents: CustomerConsent[] | null;
  publisherConsents: Consent[] | null;
  canUpdateEmail: boolean;
  canRenewSubscription: boolean;
  canChangePasswordWithOldPassword: boolean;
  setLoading: (loading: boolean) => void;
};

export const useAccountStore = createStore<AccountStore>('AccountStore', (set) => ({
  loading: true,
  auth: null,
  user: null,
  subscription: null,
  transactions: null,
  activePayment: null,
  customerConsents: null,
  publisherConsents: null,
  canUpdateEmail: false,
  canRenewSubscription: false,
  canChangePasswordWithOldPassword: false,
  setLoading: (loading: boolean) => set({ loading }),
}));
