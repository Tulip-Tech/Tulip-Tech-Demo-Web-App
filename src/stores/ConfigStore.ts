import { createStore } from './utils';

import type { AccessModel, Config } from '#types/Config';
import type { AdSchedule } from '#types/ad-schedule';

export enum PersonalShelf {
  ContinueWatching = 'continue_watching',
  Favorites = 'favorites',
}

export const PersonalShelves = [PersonalShelf.Favorites, PersonalShelf.ContinueWatching];

type CleengData = {
  cleengId: string | null | undefined;
  cleengSandbox: boolean;
  monthlyOfferId: string;
  yearlyOfferId: string;
};

type ConfigState = {
  config: Config;
  accessModel: AccessModel;
  adScheduleData: AdSchedule | null | undefined;
  getCleengData: () => CleengData;
};

export const useConfigStore = createStore<ConfigState>('ConfigStore', (_, get) => ({
  config: {
    id: '',
    siteName: '',
    description: '',
    player: '',
    assets: {},
    content: [],
    menu: [],
    integrations: {
      cleeng: {
        useSandbox: true,
      },
      jwp: {
        clientId: null,
        assetId: null,
        useSandbox: true,
      },
    },
    styling: {
      footerText: '',
    },
  },
  accessModel: 'SVOD',
  adScheduleData: null,
  getCleengData: (): CleengData => {
    const cleeng = get().config?.integrations?.cleeng;

    const cleengId = cleeng?.id;
    const cleengSandbox = !!cleeng?.useSandbox;
    const monthlyOfferId = cleeng?.monthlyOffer || '';
    const yearlyOfferId = cleeng?.yearlyOffer || '';

    return { cleengId, cleengSandbox, monthlyOfferId, yearlyOfferId };
  },
}));
