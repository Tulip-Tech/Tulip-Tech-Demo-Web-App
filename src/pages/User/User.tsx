import React, { useCallback, useEffect, useState } from 'react';
import { Navigate, Route, Routes, useNavigate } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import shallow from 'zustand/shallow';

import styles from './User.module.scss';

import PlaylistContainer from '#src/containers/PlaylistContainer/PlaylistContainer';
import { mediaURL } from '#src/utils/formatting';
import AccountCircle from '#src/icons/AccountCircle';
import Favorite from '#src/icons/Favorite';
import BalanceWallet from '#src/icons/BalanceWallet';
import Exit from '#src/icons/Exit';
import { useAccountStore } from '#src/stores/AccountStore';
import { PersonalShelf, useConfigStore } from '#src/stores/ConfigStore';
import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';
import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
import ConfirmationDialog from '#components/ConfirmationDialog/ConfirmationDialog';
import Payment from '#components/Payment/Payment';
import AccountComponent from '#components/Account/Account';
import Button from '#components/Button/Button';
import Favorites from '#components/Favorites/Favorites';
import type { PlaylistItem } from '#types/playlist';
import { logout } from '#src/stores/AccountController';
import { clear as clearFavorites } from '#src/stores/FavoritesController';

const User = (): JSX.Element => {
  const { accessModel, favoritesList } = useConfigStore(
    (s) => ({
      accessModel: s.accessModel,
      favoritesList: s.config.features?.favoritesList,
    }),
    shallow,
  );
  const navigate = useNavigate();
  const { t } = useTranslation('user');
  const breakpoint = useBreakpoint();
  const [clearFavoritesOpen, setClearFavoritesOpen] = useState(false);
  const [showAllTransactions, setShowAllTransactions] = useState(false);
  const isLargeScreen = breakpoint > Breakpoint.md;
  const { user: customer, subscription, transactions, activePayment, loading, canUpdateEmail, canRenewSubscription } = useAccountStore();

  const onCardClick = (playlistItem: PlaylistItem) => navigate(mediaURL(playlistItem));
  const onLogout = useCallback(async () => {
    // Empty customer on a user page leads to navigate (code bellow), so we don't repeat it here
    await logout();
  }, []);

  useEffect(() => {
    if (!loading && !customer) {
      navigate('/', { replace: true });
    }
  }, [navigate, customer, loading]);

  if (!customer) {
    return (
      <div className={styles.user}>
        <LoadingOverlay inline />
      </div>
    );
  }

  return (
    <div className={styles.user}>
      {isLargeScreen && (
        <div className={styles.leftColumn}>
          <div className={styles.panel}>
            <ul>
              <li>
                <Button to="my-account" label={t('nav.account')} variant="text" startIcon={<AccountCircle />} className={styles.button} />
              </li>
              {favoritesList && (
                <li>
                  <Button to="favorites" label={t('nav.favorites')} variant="text" startIcon={<Favorite />} className={styles.button} />
                </li>
              )}
              {accessModel !== 'AVOD' && (
                <li>
                  <Button to="payments" label={t('nav.payments')} variant="text" startIcon={<BalanceWallet />} className={styles.button} />
                </li>
              )}
              <li className={styles.logoutLi}>
                <Button onClick={onLogout} label={t('nav.logout')} variant="text" startIcon={<Exit />} className={styles.button} />
              </li>
            </ul>
          </div>
        </div>
      )}
      <div className={styles.mainColumn}>
        <Routes>
          <Route
            path="my-account"
            element={<AccountComponent panelClassName={styles.panel} panelHeaderClassName={styles.panelHeader} canUpdateEmail={canUpdateEmail} />}
          />
          {favoritesList && (
            <Route
              path="favorites"
              element={
                <>
                  <PlaylistContainer type={PersonalShelf.Favorites} showEmpty>
                    {({ playlist, error, isLoading }) => (
                      <Favorites
                        playlist={playlist}
                        error={error}
                        isLoading={isLoading}
                        onCardClick={onCardClick}
                        onClearFavoritesClick={() => setClearFavoritesOpen(true)}
                        accessModel={accessModel}
                        hasSubscription={!!subscription}
                      />
                    )}
                  </PlaylistContainer>
                  <ConfirmationDialog
                    open={clearFavoritesOpen}
                    title={t('favorites.clear_favorites_title')}
                    body={t('favorites.clear_favorites_body')}
                    onConfirm={async () => {
                      await clearFavorites();
                      setClearFavoritesOpen(false);
                    }}
                    onClose={() => setClearFavoritesOpen(false)}
                  />
                </>
              }
            />
          )}
          <Route
            path="payments"
            element={
              accessModel !== 'AVOD' ? (
                <Payment
                  accessModel={accessModel}
                  activeSubscription={subscription}
                  activePaymentDetail={activePayment}
                  transactions={transactions}
                  customer={customer}
                  isLoading={loading}
                  panelClassName={styles.panel}
                  panelHeaderClassName={styles.panelHeader}
                  onShowAllTransactionsClick={() => setShowAllTransactions(true)}
                  showAllTransactions={showAllTransactions}
                  canRenewSubscription={canRenewSubscription}
                />
              ) : (
                <Navigate to="my-account" />
              )
            }
          />
          <Route path="*" element={<Navigate to="my-account" />} />
        </Routes>
      </div>
    </div>
  );
};

export default User;
