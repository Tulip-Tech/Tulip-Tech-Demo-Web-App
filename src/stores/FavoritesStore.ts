import { createStore } from './utils';

import type { Favorite } from '#types/favorite';
import { PersonalShelf } from '#src/stores/ConfigStore';
import type { Playlist, PlaylistItem } from '#types/playlist';

type FavoritesState = {
  favorites: Favorite[];
  warning: string | null;
  hasItem: (item: PlaylistItem) => boolean;
  getPlaylist: () => Playlist;
  setWarning: (warning: string) => void;
  clearWarning: () => void;
};

export const useFavoritesStore = createStore<FavoritesState>('FavoritesState', (set, get) => ({
  favorites: [],
  warning: null,
  setWarning: (message: string | null) => set({ warning: message }),
  clearWarning: () => set({ warning: null }),
  hasItem: (item: PlaylistItem) => get().favorites.some((favoriteItem) => favoriteItem.mediaid === item.mediaid),
  getPlaylist: () =>
    ({
      feedid: PersonalShelf.Favorites,
      title: 'Favorites',
      playlist: get().favorites.map(({ playlistItem }) => playlistItem),
    } as Playlist),
}));
