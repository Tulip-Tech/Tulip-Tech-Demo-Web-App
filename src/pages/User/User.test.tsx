import User from './User';

import { useAccountStore } from '#src/stores/AccountStore';
import { mockWindowLocation, renderWithRouter } from '#test/testUtils';
import type { Transaction, PaymentDetail, Subscription } from '#types/subscription';
import { useConfigStore } from '#src/stores/ConfigStore';
import type { Config } from '#types/Config';
import { useFavoritesStore } from '#src/stores/FavoritesStore';
import type { Playlist, PlaylistItem } from '#types/playlist';

const data = {
  loading: false,
  user: {
    id: 'user1',
    email: 'user@jwplayer.com',
    country: 'us',
    firstName: 'User',
    lastName: 'Person',
    regDate: 'Jan 1, 2000',
    lastUserIp: '192.0.0.1',
  },
  subscription: {
    subscriptionId: 90210,
    offerId: 'offer1111',
    expiresAt: 1767994,
    status: 'active',
    nextPaymentPrice: 199.99,
    nextPaymentCurrency: 'USD',
    paymentGateway: '',
    paymentMethod: 'Paypal',
    offerTitle: 'Your Cool Subscription',
    period: 'year',
    totalPrice: 99.95,
  } as Subscription,
  transactions: [
    {
      transactionId: '11232',
      offerId: 'offer1111',
      paymentMethod: 'Ideal',
      offerTitle: 'Your Cool Subscription',
      transactionPriceInclTax: 12.88,
      transactionCurrency: 'EUR',
      customerCountry: 'nl',
      transactionDate: '1657994000',
    } as unknown as Transaction,
    {
      transactionId: '11234',
      offerId: 'offer13',
      paymentMethod: 'Cash',
      offerTitle: 'Your Cool Subscription',
      transactionPriceInclTax: 12.88,
      transactionCurrency: 'EUR',
      customerCountry: 'uk',
      transactionDate: '1667990000',
    } as unknown as Transaction,
  ],
  activePayment: {
    id: 887765,
    paymentMethodSpecificParams: {
      lastCardFourDigits: '9888',
      cardExpirationDate: '03/30',
    },
  } as unknown as PaymentDetail,
};

describe('User Component tests', () => {
  test('Account Page', () => {
    useAccountStore.setState(data);

    mockWindowLocation('/my-account');
    const { container } = renderWithRouter(<User />);

    expect(container).toMatchSnapshot();
  });

  test('Payments Page', () => {
    useAccountStore.setState(data);
    mockWindowLocation('/payments');
    const { container } = renderWithRouter(<User />);

    expect(container).toMatchSnapshot();
  });

  test('Favorites Page', () => {
    useAccountStore.setState(data);
    useConfigStore.setState({
      config: {
        features: {
          favoritesList: 'abcdefgh',
        },
        styling: {},
      } as unknown as Config,
    });
    useFavoritesStore.setState({
      getPlaylist: (): Playlist => {
        return {
          title: 'These are my favorite things',
          playlist: [
            {
              mediaid: 'aaabbbcc',
              title: 'Fav 1',
              duration: 12,
              feedid: 'abcdffff',
            } as unknown as PlaylistItem,
            {
              mediaid: 'aaabbbcd',
              title: 'Big Buck Bunny',
              duration: 6000,
              feedid: 'bbbdddff',
            } as unknown as PlaylistItem,
            {
              mediaid: 'ggaaccvv',
              title: 'My last favorite',
              duration: 659,
              feedid: 'bbbbbbbb',
            } as unknown as PlaylistItem,
          ],
        };
      },
    });
    //vi.fn().mockImplementation(() => window.location).mockReturnValue('/favorites');
    // const windowMock = vi.fn().mockReturnValue(window.location).mockReturnValue('/favorites');
    mockWindowLocation('/favorites');

    const { container } = renderWithRouter(<User />);

    expect(container).toMatchSnapshot();
  });
});
