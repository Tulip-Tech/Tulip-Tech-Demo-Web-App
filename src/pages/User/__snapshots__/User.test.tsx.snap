// Vitest Snapshot v1

exports[`User Component tests > Account Page 1`] = `
<div>
  <div
    class="_user_e9c5ca"
  >
    <div
      class="_leftColumn_e9c5ca"
    >
      <div
        class="_panel_e9c5ca"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296 _active_f8f296"
              href="/my-account"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.account
              </span>
            </a>
          </li>
          <li>
            <a
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
              href="/payments"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M5,3C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V16.72C21.59,16.37 22,15.74 22,15V9C22,8.26 21.59,7.63 21,7.28V5A2,2 0 0,0 19,3H5M5,5H19V7H13A2,2 0 0,0 11,9V15A2,2 0 0,0 13,17H19V19H5V5M13,9H20V15H13V9M16,10.5A1.5,1.5 0 0,0 14.5,12A1.5,1.5 0 0,0 16,13.5A1.5,1.5 0 0,0 17.5,12A1.5,1.5 0 0,0 16,10.5Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.payments
              </span>
            </a>
          </li>
          <li
            class="_logoutLi_e9c5ca"
          >
            <button
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M19,3H5C3.89,3 3,3.89 3,5V9H5V5H19V19H5V15H3V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10.08,15.58L11.5,17L16.5,12L11.5,7L10.08,8.41L12.67,11H3V13H12.67L10.08,15.58Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.logout
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="_mainColumn_e9c5ca"
    >
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            account.email
          </h3>
        </div>
        <div
          class="_flexBox_1c1c63"
        >
          <div
            class="_textField_e16c1b"
          >
            <label
              class="_label_e16c1b"
              for="text-field_1235_email"
            >
              account.email
            </label>
            <p>
              user@jwplayer.com
            </p>
          </div>
        </div>
        <div
          class="_controls_1c1c63"
        />
      </div>
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            account.security
          </h3>
        </div>
        <div
          class="_flexBox_1c1c63"
        >
          <strong>
            account.password
          </strong>
          <p>
            ****************
          </p>
        </div>
        <div
          class="_controls_1c1c63"
        >
          <button
            class="_button_f8f296 _default_f8f296 _outlined_f8f296"
            type="button"
          >
            <span>
              account.edit_password
            </span>
          </button>
        </div>
      </div>
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            account.about_you
          </h3>
        </div>
        <div
          class="_flexBox_1c1c63"
        >
          <div
            class="_textField_e16c1b"
          >
            <label
              class="_label_e16c1b"
              for="text-field_1235_firstName"
            >
              account.firstname
            </label>
            <p>
              User
            </p>
          </div>
          <div
            class="_textField_e16c1b"
          >
            <label
              class="_label_e16c1b"
              for="text-field_1235_lastName"
            >
              account.lastname
            </label>
            <p>
              Person
            </p>
          </div>
        </div>
        <div
          class="_controls_1c1c63"
        >
          <button
            class="_button_f8f296 _default_f8f296 _outlined_f8f296"
            type="button"
          >
            <span>
              account.edit_information
            </span>
          </button>
        </div>
      </div>
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            account.terms_and_tracking
          </h3>
        </div>
        <div
          class="_flexBox_1c1c63"
        />
        <div
          class="_controls_1c1c63"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`User Component tests > Favorites Page 1`] = `
<div>
  <div
    class="_user_e9c5ca"
  >
    <div
      class="_leftColumn_e9c5ca"
    >
      <div
        class="_panel_e9c5ca"
      >
        <ul>
          <li>
            <a
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
              href="/my-account"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.account
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296 _active_f8f296"
              href="/favorites"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.favorites
              </span>
            </a>
          </li>
          <li>
            <a
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
              href="/payments"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M5,3C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V16.72C21.59,16.37 22,15.74 22,15V9C22,8.26 21.59,7.63 21,7.28V5A2,2 0 0,0 19,3H5M5,5H19V7H13A2,2 0 0,0 11,9V15A2,2 0 0,0 13,17H19V19H5V5M13,9H20V15H13V9M16,10.5A1.5,1.5 0 0,0 14.5,12A1.5,1.5 0 0,0 16,13.5A1.5,1.5 0 0,0 17.5,12A1.5,1.5 0 0,0 16,10.5Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.payments
              </span>
            </a>
          </li>
          <li
            class="_logoutLi_e9c5ca"
          >
            <button
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M19,3H5C3.89,3 3,3.89 3,5V9H5V5H19V19H5V15H3V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10.08,15.58L11.5,17L16.5,12L11.5,7L10.08,8.41L12.67,11H3V13H12.67L10.08,15.58Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.logout
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="_mainColumn_e9c5ca"
    >
      <div>
        <div
          class="_header_546956"
        >
          <h3>
            favorites.title
          </h3>
          <button
            class="_button_f8f296 _default_f8f296 _outlined_f8f296"
          >
            <span>
              favorites.clear
            </span>
          </button>
        </div>
        <div>
          <div
            class="_container_7175d1 _cols-3_7175d1"
            role="grid"
          >
            <div
              class="_cell_7175d1"
              role="row"
            >
              <div
                role="cell"
              >
                <div
                  aria-label="play_item"
                  class="_card_d75732"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="_poster_d75732 _aspect169_d75732"
                  >
                    <div
                      class="_meta_d75732"
                    >
                      <div
                        class="_tags_d75732"
                      >
                        <div
                          class="_tag_d75732"
                        >
                          1 min
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="_titleContainer_d75732"
                  >
                    <div
                      class="_title_d75732"
                    >
                      Fav 1
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="_cell_7175d1"
              role="row"
            >
              <div
                role="cell"
              >
                <div
                  aria-label="play_item"
                  class="_card_d75732"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="_poster_d75732 _aspect169_d75732"
                  >
                    <div
                      class="_meta_d75732"
                    >
                      <div
                        class="_tags_d75732"
                      >
                        <div
                          class="_tag_d75732"
                        >
                          100 min
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="_titleContainer_d75732"
                  >
                    <div
                      class="_title_d75732"
                    >
                      Big Buck Bunny
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="_cell_7175d1"
              role="row"
            >
              <div
                role="cell"
              >
                <div
                  aria-label="play_item"
                  class="_card_d75732"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="_poster_d75732 _aspect169_d75732"
                  >
                    <div
                      class="_meta_d75732"
                    >
                      <div
                        class="_tags_d75732"
                      >
                        <div
                          class="_tag_d75732"
                        >
                          11 min
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="_titleContainer_d75732"
                  >
                    <div
                      class="_title_d75732"
                    >
                      My last favorite
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`User Component tests > Payments Page 1`] = `
<div>
  <div
    class="_user_e9c5ca"
  >
    <div
      class="_leftColumn_e9c5ca"
    >
      <div
        class="_panel_e9c5ca"
      >
        <ul>
          <li>
            <a
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
              href="/my-account"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M12,19.2C9.5,19.2 7.29,17.92 6,16C6.03,14 10,12.9 12,12.9C14,12.9 17.97,14 18,16C16.71,17.92 14.5,19.2 12,19.2M12,5A3,3 0 0,1 15,8A3,3 0 0,1 12,11A3,3 0 0,1 9,8A3,3 0 0,1 12,5M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.account
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296 _active_f8f296"
              href="/payments"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M5,3C3.89,3 3,3.9 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V16.72C21.59,16.37 22,15.74 22,15V9C22,8.26 21.59,7.63 21,7.28V5A2,2 0 0,0 19,3H5M5,5H19V7H13A2,2 0 0,0 11,9V15A2,2 0 0,0 13,17H19V19H5V5M13,9H20V15H13V9M16,10.5A1.5,1.5 0 0,0 14.5,12A1.5,1.5 0 0,0 16,13.5A1.5,1.5 0 0,0 17.5,12A1.5,1.5 0 0,0 16,10.5Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.payments
              </span>
            </a>
          </li>
          <li
            class="_logoutLi_e9c5ca"
          >
            <button
              class="_button_f8f296 _button_e9c5ca _default_f8f296 _text_f8f296"
            >
              <div
                class="_startIcon_f8f296"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_1e9999"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M19,3H5C3.89,3 3,3.89 3,5V9H5V5H19V19H5V15H3V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M10.08,15.58L11.5,17L16.5,12L11.5,7L10.08,8.41L12.67,11H3V13H12.67L10.08,15.58Z"
                    />
                  </g>
                </svg>
              </div>
              <span>
                nav.logout
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="_mainColumn_e9c5ca"
    >
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            user:payment.subscription_details
          </h3>
        </div>
        <div
          class="_infoBox_c57ff5"
        >
          <p>
            <strong>
              user:payment.annual_subscription
            </strong>
             
            <br />
            user:payment.next_billing_date_on
          </p>
          <p
            class="_price_c57ff5"
          >
            <strong>
              $199.99
            </strong>
            <small>
              /
              account:periods.year
            </small>
          </p>
        </div>
        <button
          class="_button_f8f296 _default_f8f296 _outlined_f8f296"
        >
          <span>
            user:payment.cancel_subscription
          </span>
        </button>
      </div>
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            user:payment.payment_method
          </h3>
        </div>
        <div>
          <div
            class="_textField_e16c1b"
          >
            <label
              class="_label_e16c1b"
              for="text-field_1235"
            >
              user:payment.card_number
            </label>
            <p>
              •••• •••• •••• 9888
            </p>
          </div>
          <div
            class="_cardDetails_c57ff5"
          >
            <div
              class="_textField_e16c1b"
            >
              <label
                class="_label_e16c1b"
                for="text-field_1235"
              >
                user:payment.expiry_date
              </label>
              <p>
                03/30
              </p>
            </div>
            <div
              class="_textField_e16c1b"
            >
              <label
                class="_label_e16c1b"
                for="text-field_1235"
              >
                user:payment.cvc_cvv
              </label>
              <p>
                ******
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="_panel_e9c5ca"
      >
        <div
          class="_panelHeader_e9c5ca"
        >
          <h3>
            user:payment.transactions
          </h3>
        </div>
        <div
          class="_infoBox_c57ff5"
        >
          <p>
            <strong>
              Your Cool Subscription
            </strong>
             
            <br />
            user:payment.price_payed_with
          </p>
          <p>
            11232
            <br />
            7/16/2022
          </p>
        </div>
        <div
          class="_infoBox_c57ff5"
        >
          <p>
            <strong>
              Your Cool Subscription
            </strong>
             
            <br />
            user:payment.price_payed_with
          </p>
          <p>
            11234
            <br />
            11/9/2022
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
